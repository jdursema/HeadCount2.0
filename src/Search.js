import React, { Component } from 'react';
import './style/search.css'

const Search = ({search}) => {
  return(
    <input placeholder='Search' onChange={(event) => search(event.target.value)}/>
  )
}
export default Search;