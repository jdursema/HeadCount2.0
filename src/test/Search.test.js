import React from 'react';
import { shallow, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import Search from '../Search';

describe('Search Test', () => {

  it('should be defined', () => {
    const search = shallow(<Search />);

    expect(search).toBeDefined();

  });

  it('should receive props', () => {
    const mockFunc = jest.fn();
    const search = shallow(<Search search={mockFunc} />);

    expect(search.instance().props.search).toEqual(mockFunc);

  });

  it('should match snap shot', () => {
    const mockFunc = jest.fn();
    const search = shallow(<Search search={mockFunc} />);

    expect(search).toMatchSnapshot()
  });

});

