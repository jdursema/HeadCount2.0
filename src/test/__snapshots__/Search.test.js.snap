exports[`Search Test should match snap shot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Search
    search={[Function]} />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <input
        onChange={[Function]}
        placeholder="Search" />,
      "className": "search-field",
    },
    "ref": null,
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "onChange": [Function],
        "placeholder": "Search",
      },
      "ref": null,
      "rendered": null,
      "type": "input",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <input
          onChange={[Function]}
          placeholder="Search" />,
        "className": "search-field",
      },
      "ref": null,
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "onChange": [Function],
          "placeholder": "Search",
        },
        "ref": null,
        "rendered": null,
        "type": "input",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactFifteenAdapter {
      "options": Object {
        "supportPrevContextArgumentOfComponentDidUpdate": true,
      },
    },
  },
}
`;
